pkg/process/checks/process_events_linux.go

```diff
@@ -0,0 +1,187 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package checks
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	payload "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/DataDog/datadog-agent/pkg/process/config"
+	"github.com/DataDog/datadog-agent/pkg/process/events"
+	"github.com/DataDog/datadog-agent/pkg/process/events/model"
+	"github.com/DataDog/datadog-agent/pkg/process/statsd"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ProcessEvents is a ProcessEventsCheck singleton
+var ProcessEvents = &ProcessEventsCheck{}
+
+// ProcessEventsCheck collects process lifecycle events such as exec and exit signals
+type ProcessEventsCheck struct {
+	store    events.Store
+	listener *events.SysProbeListener
+	sysInfo  *payload.SystemInfo
+
+	maxBatchSize int
+}
+
+// Init initializes the ProcessEventsCheck.
+func (e *ProcessEventsCheck) Init(_ *config.AgentConfig, info *payload.SystemInfo) {
+	log.Info("Initializing process_events check")
+	e.sysInfo = info
+	e.maxBatchSize = getMaxBatchSize()
+
+	store, err := events.NewRingStore(statsd.Client)
+	if err != nil {
+		log.Errorf("RingStore can't be created: %v", err)
+		return
+	}
+	e.store = store
```

Comment:
Would it make sense to check if the instance was already initialized, just in case?

Reply:
Normally it can't happen that the collector calls `Init` more than once. But I agree that it would be better to prevent it from happening. I've updated it on https://github.com/DataDog/datadog-agent/pull/12390/commits/fbd8c8bdaf01f0316c18ec52cef448fb49dcd395.